testsuite/tests/typing-local/let_mutable.ml

```diff
@@ -0,0 +1,385 @@
+(* TEST
+   flags = "-extension let_mutable";
+   include stdlib_upstream_compatible;
+   expect; *)
+
+(* Test 1.1: basic usage in a for loop *)
+let foo1 y =
+  let mutable x = y in
+  for i = 1 to 10 do
+    x <- x + i
+  done;
+  (x : int)
```

Comment:
I think this type annotation, and the ones like it on similar examples below, are no longer needed.